[docs][yba] Control Plane RBAC with custom roles (MVP) (#19588)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/anywhere-rbac.md

    identifier: anywhere-rbac
    weight: 10
type: docs
---

YugabyteDB Anywhere uses a role-based access control (RBAC) model to manage access to your YugabyteDB Anywhere instance (whether via the UI or the REST API). Using roles, you can enforce the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) (PoLP) by ensuring that users have the precise permissions needed to fulfill their roles while mitigating the risk of unauthorized access or accidental breaches. A role defines a set of permissions that determine what features can be accessed by account users who have been assigned that role.

## Users

A user can interact with a YugabyteDB Anywhere through the UI or REST API.

Users are assigned [roles](#roles), which define the set of actions users can perform. You can also define a set of universes to which the user has access.

### Create, modify, and delete users

As a Super Admin or Admin, you can invite new users and manage existing users for your YugabyteDB Anywhere instance.

You can assign built-in and custom roles to users to determine the actions they are allowed to perform, and specify the universes that they can access.

> To create a user, do the following:

Reviewer: Would it be better to add images here to help guide the user better?

### Built-in roles

The following built-in roles are available:

* **Super Admin** is the first user that is created during the installation. This role has the highest level of privilege that allows all read and write actions on all YugabyteDB Anywhere resources. There can be only one Super Admin. This Super Admin can perform the following:

  * Manage all resources, including universes, nodes, backup, restore, and cloud providers.
  * Manage the user access control by creating and managing users.

  For more information, see [admin user](../../configure-yugabyte-platform/create-admin-user/).

* **Admin** has privileges that are similar to the Super Admin, except that the Admin cannot manage the global scope artifacts and actions, such as runtime configuration settings and LDAP authentication.

* **Backup Admin** has access to the backup related tasks, such as the following:

  * Manage database backups and restore operations.
  * Create new backups.
  * Delete any existing backup or backup schedule.
  * Edit existing backups.
  * Read-only permissions for all other resources in YugabyteDB Anywhere.

> * **Read-only** access level provides viewer permission to the UI and API.

Reviewer: Theres another built-in role called "Connect Only".

To create a custom role, do the following:

1. Navigate to **Admin > Access Management > Roles** and click **Create Role**.
1. Enter a name for the role.
1. Enter a description for the role.
1. Click **Select Permissions**.
1. Select the permissions to assign to the role and click **Confirm** when you are done.
1. Click **Save**.

To create a custom role from an existing role, do the following:

1. Navigate to **Admin > Access Management > Roles**, click **Actions** for the role to clone, and choose **Clone Role**.
1. Enter a name for the role.
1. Enter a description for the role.
1. Click **Edit Permissions**.
1. Select the permissions to assign to the role and click **Confirm** when you are done.
1. Click **Save**.

To edit a custom role, do the following:

1. Navigate to **Admin > Access Management > Roles**, click **Actions** for the role to edit, and choose **Edit Role**.
1. On the **Configurations** tab click **Edit Permissions**.
1. Select the permissions to assign to the role and click **Confirm** when you are done.
1. Click **Save**.

To delete a role, navigate to **Admin > Access Management > Roles**, click **Actions** for the role to edit, and choose **Edit Role**.

> To view the users that have been assigned a role, navigate to **Admin > Access Management > Roles**, click **Actions** for the role, and choose **View Assigned Users**.

Reviewer: @ddhodge @vpatibandla-yb Should we also add here some of the limitations (from our internal design doc) that we have in RBAC phase 1, which will be solved in phase 2?

---
title: Manage access to YugabyteDB Anywhere
headerTitle: Manage YugabyteDB Anywhere users
linkTitle: Manage users
description: Use role-based access control (RBAC) in YugabyteDB Anywhere to manage users and roles.
headcontent: Invite team members to your account and manage their access
menu:
  preview_yugabyte-platform:
    parent: administer-yugabyte-platform
    identifier: anywhere-rbac
    weight: 27
type: docs
---

YugabyteDB Anywhere uses a role-based access control (RBAC) model to manage access to your YugabyteDB Anywhere instance (whether via the UI or the REST API). Using roles, you can enforce the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) (PoLP) by ensuring that users have the precise permissions needed to fulfill their roles while mitigating the risk of unauthorized access or accidental breaches. A role defines a set of permissions that determine what features can be accessed by account users who have been assigned that role.

## Users

A user can interact with a YugabyteDB Anywhere through the UI or REST API.

> Users are assigned roles, which determine what they actions they can perform.

Reviewer 2: which determine what are the actions they can perform.

> Users are assigned roles, which determine what they actions they can perform.

Author: ```suggestion
Users are assigned roles, which define the set of actions users can perform.
```

  * Manage all resources, including universes, nodes, backup, restore, and cloud providers.
  * Manage the user access control by creating and managing users.

  For more information, see [admin user](../../configure-yugabyte-platform/create-admin-user/).

* **Admin** has privileges that are similar to the Super Admin, except that the Admin cannot manage the global scope artifacts and actions, such as runtime configuration settings and LDAP authentication.

* **Backup Admin** has access to the backup related tasks, such as the following:

  * Manage database backups and restore operations.
  * Create new backups.
  * Delete any existing backup or backup schedule.
  * Edit existing backups.
  * Read-only permissions for all other resources in YugabyteDB Anywhere.

* **Read-only** access level provides viewer permission to the UI and API.

### Create, modify, and delete roles

As a Super Admin or Admin, you can create and modify roles.

To create a custom role, do the following:

> 1. Navigate to **Admin > User Management > Roles**, then click **Add Role** to display the **Create a Role** dialog.

Reviewer 2: Admin > Access Management > Roles

Reviewer 2: With the new RBAC, we can not only create Custom Roles but also scope the customer roles at the universe level.

Please also include the flow to modify the role bindings of the users from the Admin > Access Management > Users > Edit Assigned Roles option for the user.

Reviewer 2: The new RBAC feature which allows custom roles can be turned on by using the runtime flag yb.rbac.use_new_authz. This is an internal Runtime config as of now. I will make the change to PUBLIC once I get the QA approval.

The runtime config can be edited from the UI from the Admin > Advanced > Global Configuration section and the runtime config key yb.rbac.use_new_authz should be set to true for the user to use the new RBAC logic. Once the runtime config is set to true, it cannot be turned off

Reviewer 2: Please write a section on how to assign universes for Custom roles

Reviewer 2: Phase 1 MVP: The resource that we deal with are Universes. Later phases will deal with other kind of resources

1. Enter a name for the role.
1. Enter a description for the role.
1. Click **Select Permissions**.
1. Select the permissions to assign to the role and click **Select** when you are done.
1. Click **Save**.

> To modify a user role or delete a role, navigate to **Admin > User Management > Roles**. Click **Actions** that corresponds to the specific role, and then select either **Edit Role** or **Delete Role**.

Reviewer 2: This is the old flow, do we want to include the new RBAC workflows too?

Author: We want only the new flow I would have thought
